### PR TITLE
Remove the makefile

### PR DESCRIPTION
**Pull request by pbartholomew08**
Todo:
- [x] port CI to Cmake
- [x] add instructions for driving build/linking from an external Makefile

### CODE REVIEW

File: .github/workflows/Build.yml [hunk 1]
```diff
@@ -24,11 +24,15 @@ jobs:
...
         sudo apt-get update
-        sudo apt-get install -y gfortran openmpi-bin libopenmpi-dev
+        sudo apt-get install -y gfortran openmpi-bin libopenmpi-dev cmake
 
     # Runs a set of commands using the runners shell
     - name: Compile
-      run: make BUILD=dev
+      run: |
+        FC=mpif90 cmake -B build -DCMAKE_BUILD_TYPE=dev
+        make -C build decomp2d
```
Comment (mathrack) [inline .github/workflows/Build.yml:33]: Should be cmake ?

File: .github/workflows/Build_all.yml [hunk 2]
```diff
@@ -99,15 +103,21 @@ jobs:
...
+      run: |
+        source /opt/intel/oneapi/setvars.sh
+        FC=mpiifort cmake -B build -DCMAKE_BUILD_TYPE=debug
+        make -C build decomp2d
 
     # Runs a set of commands using the runners shell
     - name: Compile
-      run: source /opt/intel/oneapi/setvars.sh && make clean && make CMP=intel FFT=mkl BUILD=debug && make examples
+      run: |
+        source /opt/intel/oneapi/setvars.sh
```
Comment (mathrack) [inline .github/workflows/Build_all.yml:118]: Could be best to keep the make clean here ?

**Comment by p-costa** [comment]
I guess that I will have to adopt `cmake` for building 2decomp if I want to keep my Git Submodule in sync with the upstream developments?

**Comment by mathrack** [comment]
Hello @p-costa , this is an open-source project. We decided to switch to CMake some time age. Keeping up-to-date both Makefile and CMake leads to extra work...

However, a dedicated branch with support of Makefile could be maintained. If you want to take a share of this extra work, I am willing to contribute. Could be easier to maintain if restricted to CPU build, we can talk about it.

Cheers

**Comment by p-costa** [comment]
Hi Cédric, @mathrack. Thanks!

I didn't want my query to come across as imposing or anything. Worst case, if I don't adopt `cmake`, I'll keep and maintain my own `make` workflow in my fork. Anyway, I'd be happy to discuss if you plan to keep `make` and `cmake` build workflow in parallel.

**Comment by mathrack** [comment]
Sounds good, this is not merged yet, we will chat about it when it gets merged.

File: .github/workflows/Build.yml (hunk 1 above)
Comment (pbartholomew08) [reply .github/workflows/Build.yml:33]: Yes, we should follow the style from the README. Will fix

**Comment by p-costa** [comment]
Is just that it will require some exercise on our end to integrate it in codes using CaNS as base. There the 2decomp build is connected to the build profile of CaNS.

File: .github/workflows/Build_all.yml (hunk 2 above)
Comment (pbartholomew08) [reply .github/workflows/Build_all.yml:121]: Addressed by fbc5660

File: .github/workflows/Build.yml (hunk 1 above)
Comment (pbartholomew08) [reply .github/workflows/Build.yml]: Addressed by 75bd117

**Comment by pbartholomew08** [comment]
All tasks in CI have been updated to use cmake, I can run each task by hand (GNU build, Nvidia build, Intel build, MKL, FFTW3, FFTW3_F03 and caliper) and the tests pass so I assume they will continue to pass when the CI switches over with this merge (though that will be the true test)

**Review by mathrack — APPROVED** [review]

**Pull request merged**